integration-tests/tests/client/sharding_upgrade.rs

```diff
@@ -179,6 +179,55 @@ impl TestShardUpgradeEnv {
         }
     }
 
+    fn check_next_block_with_new_chunk(&mut self, height: BlockHeight) {
+        let block = self.env.clients[0].chain.get_block_by_height(height).unwrap().clone();
+        let block_hash = block.hash();
+        let num_shards = block.chunks().len();
+        for shard_id in 0..num_shards {
+            let mut last_block_hash_with_empty_chunk = match self.env.clients[0]
```

Comment:
nit: this name (last_block_hash_with_empty_chunk) is not necesarily true, as it might be equal to block_hash..

Reply:
I added more comments, hopefully that will make the meaning of this variable more clear.